File: judge/java/tests/two_difference.java
package tests;
import java.util.*;
import java.lang.*;
import java.io.*;
import judge.*;
import datastruct.*;

public class two_difference {
    public static int num_test = 90;
    public static int[][] in_0;
    public static int[][] in_org_0;
    public static int[] in_1;
    public static int[] in_org_1;
    public static int[][] out;
    public static boolean test(int[] answer, int i) {
        if(out[i].length != answer.length) return false;
        if(out[i][0] == -1 && out[i][1] == -1) {
            return answer[0] == -1 && answer[1] == -1;
        } else {
            if(answer[0] == -1 || answer[1] == -1) {
                return false;
            } else {
                return Math.abs(in_0[i][answer[0]] - in_0[i][answer[1]]) == in_1[i];
            }
        }
    }

    public static void load_test() {
        File fil = new File("judge/tests/two-difference.txt");
        FileReader inputFil = null;
        try {
            inputFil = new FileReader(fil);
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        }
        BufferedReader in = new BufferedReader(inputFil);
        try {
            in_0 = common.read_int_matrix(in);
            in_org_0 = test_common.copy(in_0);
            in_1 = common.read_int_array(in);
            in_org_1 = test_common.copy(in_1);
            out = common.read_int_matrix(in);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static void judge() {
        load_test();
        common.capture_stdout();
        Solution s = new Solution();

        long startTime = System.currentTimeMillis();

        for(int i = 0; i < num_test; ++i) {
            System.out.printf("Testing case #%d\n", i+1);
            int[] answer = s.two_dif(in_0[i], in_1[i]);
            if(!two_difference.test(answer,i)) {
                common.release_stdout();
                System.out.printf("%d / %d;", i+1, num_test);
                String outs = common.to_string(two_difference.in_org_0[i]) + ", " + common.to_string(two_difference.in_org_1[i]);
                System.out.print(outs + ";");
                System.out.print(common.to_string(answer) + ";");
                System.out.println(common.to_string(out[i]));
                return;
            }
        }

        common.release_stdout();
        long estimatedTime = System.currentTimeMillis() - startTime;
        System.out.print("Accepted;");
        System.out.println(estimatedTime);
    }
}

File: judge/lua/tests/two-difference.lua
require("../solution")

local num_test = 90
local in_0 = {}
local in_org_0 = {}
local in_1 = {}
local in_org_1 = {}
local out = {}

function test(answer, i)
    if out[i].length ~= answer.length then
        return false
    end
    if out[i][1] == -1 and out[i][2] == -1 then
        return answer[1] == -1 and answer[2] == -1
    else
        if answer[1] == -1 or answer[2] == -1 then
            return false
        else
            return math.abs(in_0[i][answer[1]] - in_0[i][answer[2]]) == in_1[i]
        end
    end
end

function load_test()
    local f = io.open("./judge/tests/two-difference.txt", "r")
    in_0 = read_num_matrix(f)
    in_org_0 = copy(in_0)
    in_1 = read_num_array(f)
    in_org_1 = copy(in_1)
    out = read_num_matrix(f)
    f:close()
end

function judge()
    load_test()
    capture_stdout()

    local start = os.clock()
    for i = 1, num_test do
        print("Testing case #" .. i)
        local answer = two_dif(in_0[i], in_1[i]) 
        if not test(answer, i) then
            io.write(string.format("%d / %d;", i, num_test))
            io.write(to_string(in_org_0[i]))
            io.write(", ")
            io.write(to_string(in_org_1[i]))
            io.write(";")
            io.write(to_string(answer))
            io.write(";")
            io.write(to_string(out[i]))
            io.write("\n")
            return
        end
    end

    release_stdout()
    local elapsed = math.floor((os.clock() - start) * 1000)
	print("Accepted;" .. elapsed)
end
